src/components/navigation/section-document-or-browser.tsx

```diff
@@ -172,9 +184,95 @@ export const SectionDocumentOrBrowser: React.FC<IProps> = observer(function Sect
       tabsExtraClassNames={{"chat-open": isChatOpen}}
       onSelect={handleTabSelect}
       selectedIndex={subTabIndex}
-      renderSubTabPanel={subTab =>
-        renderDocumentView(subTab) || renderDocumentBrowserView(subTab)
-      }
+      renderSubTabPanel={subTab => renderDocumentView(subTab) || renderDocumentBrowserView(subTab)}
     />
   );
 });
+
+interface DocumentBrowserScrollerProps {
+  subTab: ISubTabSpec;
+  tabSpec: NavTabModelType;
+  openDocumentKey: string;
+  onSelectDocument: (document: DocumentModelType) => void;
+}
+const DocumentBrowserScroller =
+    ({subTab, tabSpec, openDocumentKey, onSelectDocument}: DocumentBrowserScrollerProps) => {
+  const [scrollerCollapsed, setScrollerCollapsed] = useState(false);
+  const [panelWidth, setPanelWidth] = useState(0);
+  const [scrollWidth, setScrollWidth] = useState(0);
+  const [scrollLeft, setScrollLeft] = useState(0);
+  const documentScrollerRef = useRef<HTMLDivElement>(null);
+  const [scrollToLocation, setScrollToLocation] = useState(0);
+
+  useEffect(()=>{
+    const scrollerEl = documentScrollerRef.current;
+    if (scrollerEl) {
+      setPanelWidth(scrollerEl.getBoundingClientRect().width);
+    }
+  },[documentScrollerRef]);
+
+  const handleScrollTo = (side: string) => {
+    const direction = side ==="left" ? -1 : 1;
+    setScrollToLocation((prevState) => {
+      const tempScrollTo = prevState + (direction * panelWidth);
+      if (tempScrollTo === 0) {
```

Comment:
I want to double check the logic here as well. It seems like this check should be to make sure we're not going too far left or right? If that's the case, I'd expect something like this:
```
if (tempScrollTo < 0) {
  return 0;
} else if (tempScrollTo > scrollWidth - panelWidth) { // Here I'm assuming scrollWidth is the total width of the list, while panelWidth is the width of a single view
  return scrollWidth - panelWidth;
} else {
  return tempScrollTo;
}
```

Reply:
@tealefristoe I tried your logic and `tempScroll < 0` prevents it from scrolling backwards. Did rework some of the logic around `prevState` so it no longer goes beyond the bounds of `scrollWidth`.